x/staking/genesis.go

```diff
@@ -190,11 +196,16 @@ func ExportGenesis(ctx sdk.Context, keeper keeper.Keeper) *types.GenesisState {
 }
 
 // WriteValidators returns a slice of bonded genesis validators.
-func WriteValidators(ctx sdk.Context, keeper keeper.Keeper) (vals []tmtypes.GenesisValidator) {
+func WriteValidators(ctx sdk.Context, keeper keeper.Keeper) (vals []tmtypes.GenesisValidator, err error) {
 	keeper.IterateLastValidators(ctx, func(_ int64, validator types.ValidatorI) (stop bool) {
+		var consPk crypto.PubKey
+		consPk, err = validator.TmConsPubKey()
+		if err != nil {
+			return true
```

Comment:
Is stopping the iterator an OK behavior here? I would have thought to expect an error return, or panic

Reply:
yes, we capture the error outside of the closure and return it there.  The closure manipulates a variables from the external function context.

Reply:
So, we shouldn't panic here.